runtime/protocol/aws-json-protocols/common/src/aws/smithy/kotlin/runtime/awsprotocol/json/RestJsonErrorDeserializer.kt

```diff
@@ -47,19 +47,14 @@ public object RestJsonErrorDeserializer {
 
     public fun deserialize(headers: Headers, payload: ByteArray?): ErrorDetails {
         var code: String? = headers[X_AMZN_ERROR_TYPE_HEADER_NAME]
-        var message: String? = headers[X_AMZN_ERROR_MESSAGE_HEADER_NAME]
-        if (message == null) {
-            message = headers[X_AMZN_EVENT_ERROR_MESSAGE_HEADER_NAME]
-        }
+        var message: String? = if (headers[X_AMZN_ERROR_MESSAGE_HEADER_NAME] == null) headers[X_AMZN_EVENT_ERROR_MESSAGE_HEADER_NAME] else headers[X_AMZN_ERROR_MESSAGE_HEADER_NAME]
```

Comment:
Nice, also I used var since it can get rewritten in lines [58-61](https://github.com/awslabs/smithy-kotlin/pull/865/files#diff-5a4f973c35e932aa860bbb69df2fa65fb0cf258c2c63de9592e371d456ef1e51R58-R61). That was how the code was before I touched it, is that correct behavior or should I remove it?

Reply:
Oh you're right, I missed that. Nevermind about `val` in this case then.